### PR TITLE
[INFINITY-2236] Verify Kafka upgrade instructions from 1.1.26-beta 

### PR DESCRIPTION
**Pull request by mbdcos**
This needs to be verified manually to ensure all these instructions are up to date and correct.

->

Originally  a **generic** document was prepared,  explaining upgrade from _0.10.2 Kafka_ to _0.11.1 Kafka_, and explaining upgrade in strict mode with service account change. It was referring versions as **old version** (with x) and **new version** (with y). 

Then based on reviewers, _version numbers_ in the current document were added. Now the existing document is not a generic document. It became version specific. 
See the following PRs:
#1449
#1429

2.0.0 GA release does not have any `upgradesFrom` field, so this document will be misleading
 We do not support upgrade from betas to GA for the moment.

This PR is to modify the upgrade document from 1.1.26-beta to 1.1.27-beta (these are the only ones with `upgradesFrom` in universe at this moment for Kafka).

**Note:** There is no upgrade from DC/OS Kafka 1.1.19 to 2.0.0 GA (manual marathon upgrade). 2.0.1 is not released yet.

**Note:** This PR is only for DC/OS 1.10 upgrade. DC/OS 1.9 needs manual marathon upgrade

### CODE REVIEW

**Comment by mbdcos** [comment]
Are we going to have `upgradesFrom` field in `2.0.0-x.x.x` releases? This upgrade document is tested via adding `upgradesFrom` and making a stub universe release when it was prepared. That was the plan at that time, to upgrade from old to new.  If we are not upgrading from beta to release, that this upgrade document is obsolete now.

**Comment by nickbp** [comment]
The document isn't obsolete for upgrading between stable releases or between beta releases, it just wouldn't be an available option for moving from beta to 2.0.0. Once we have a 2.0.1 then this would be useful for upgrading from 2.0.0.

**Comment by mbdcos** [comment]
@nick only  1.1.23-beta to 1.1.24-beta  upgrade 
or
1.1.23-beta to 2.0.0 GA (now we don't support) upgrade

has these requirements due to Kafka log format change and service account config change.

1.1.24 to 2.0.0 upgrade does not have these requirements.
Also, 2.0.0 to 2.0.1 does not have these requirements.

Let's not keep documented instructions if we know will not work. Let's update it or remove it.

@gabriel Can you review frameworks/docs/upgrade.md?

Originally I prepared a generic document explaining upgrade from 0.10.2 Kafka to 0.11.1 Kafka, and explaining upgrade in strict mode with service account change. I was referring versions as old version (with x) and new version (with y). 

Then based on reviewers, version numbers in the current document were added. Now it is not a generic document. It became version specific. 

Either let's make it beta to beta upgrade or remove this doc. My concern is that will be confusing if we can not upgrade from beta to GA.

There is no upgrade from 1.1.19 to 2.0.0 (manual). 2.0.1 is not released yet.

Any comments. I am inclined to remove this upgrade.md document.

**Comment by mbdcos** [comment]
Upgrade from 1.1.26 beta-to 1.1.27-beta (upgradesFrom is available only between these two at this moment)
--

File: frameworks/kafka/docs/upgrade.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+post_title: Upgrade `1.1.26-0.10.1.0-beta` to `1.1.27-0.11.0-beta`
 feature_maturity: preview
 enterprise: 'no'
 ---
 
-This document explains how to upgrade the DC/OS Apache Kafka service from version `1.1.16-0.10.1.0-beta` to `2.0.0-0.11.0.0`. Upgrading an existing DC/OS Apache Kafka service in `strict mode` requires special attention. Refer to [Upgrade in Strict Mode](#upgrade-in-strict-mode) for more information.
+This document explains how to upgrade the DC/OS Apache Kafka service from version `1.1.26-0.10.1.0-beta` to `1.1.27-0.11.0-beta`. Upgrading an existing DC/OS Apache Kafka service in `strict mode` requires special attention. Refer to [Upgrade in Strict Mode](#upgrade-in-strict-mode) for more information.
 
-##  Upgrade Kafka from 0.10.2.x to 0.11.0.0
+##  Upgrade beta-kafka from 0.10.2.x to 0.11.0.0
```
Comment (gabrielhartmann) [inline frameworks/kafka/docs/upgrade.md]: We should not reference version numbers that refer only to base-tech.

File: frameworks/kafka/docs/upgrade.md [hunk 2]
```diff
@@ -1,12 +1,12 @@
 ---
-post_title: Upgrade `1.1.16-0.10.1.0-beta` to `2.0.0-0.11.0.0`
+post_title: Upgrade `1.1.26-0.10.1.0-beta` to `1.1.27-0.11.0-beta`
 feature_maturity: preview
 enterprise: 'no'
 ---
 
-This document explains how to upgrade the DC/OS Apache Kafka service from version `1.1.16-0.10.1.0-beta` to `2.0.0-0.11.0.0`. Upgrading an existing DC/OS Apache Kafka service in `strict mode` requires special attention. Refer to [Upgrade in Strict Mode](#upgrade-in-strict-mode) for more information.
+This document explains how to upgrade the DC/OS Apache Kafka service from version `1.1.26-0.10.1.0-beta` to `1.1.27-0.11.0-beta`. Upgrading an existing DC/OS Apache Kafka service in `strict mode` requires special attention. Refer to [Upgrade in Strict Mode](#upgrade-in-strict-mode) for more information.
```
Comment (sascala) [inline frameworks/kafka/docs/upgrade.md]: Upgrading an existing Kafka service on a [strict security mode cluster](https://docs.mesosphere.com/1.9/security/#security-modes) requires an extra step.

File: frameworks/kafka/docs/upgrade.md [hunk 3]
```diff
@@ -38,24 +38,24 @@ This is an overview of the rolling upgrade process. The process for upgrading Ka
 
 ### Detailed upgrade instructions
 
-The new DC/OS Kafka package `2.0.0-0.11.0.0` (with Kafka version 0.11.0.0) that we will be upgrading to has the following default settings:
+The new DC/OS Kafka package `1.1.27-0.11.0-beta` (with Kafka version 0.11.0.0) that we will be upgrading to has the following default settings:
```
Comment (sascala) [inline frameworks/kafka/docs/upgrade.md]: The new DC/OS Kafka package `1.1.27-0.11.0-beta` (with Kafka version 0.11.0.0) has the following default settings

File: frameworks/kafka/docs/upgrade.md [hunk 4]
```diff
@@ -38,24 +38,24 @@ This is an overview of the rolling upgrade process. The process for upgrading Ka
...
-The beta DC/OS Kafka package `1.1.16-0.10.1.0-beta` (with Kafka version 0.10.2.x) has the following configuration:
+The beta DC/OS Kafka package `1.1.26-0.10.1.0-beta` (with Kafka version 0.10.2.x) has the following configuration:
 
 * Kafka Version = 0.10.2.1
 * inter.broker.protocol.version = 0.10.0.0
 * log.message.format.version = 0.10.0
 
 
-Here, we assume that you already have a DC/OS Kafka service running, installed with the beta package version `1.1.16-0.10.1.0-beta` (with Kafka 0.10.2.x). In order to guarantee no downtime during the upgrade process, follow the steps below.  
+Here, we assume that you already have a DC/OS Kafka service running, installed with the beta package version `1.1.26-0.10.1.0-beta` (with Kafka 0.10.2.x). In order to guarantee no downtime during the upgrade process, follow the steps below.
```
Comment (sascala) [inline frameworks/kafka/docs/upgrade.md]: Here, we assume you already have beta DC/OS Kafka version `1.1.26-0.10.1.0-beta` (with Kafka 0.10.2.x) installed.

File: frameworks/kafka/docs/upgrade.md [hunk 5]
```diff
@@ -38,24 +38,24 @@ This is an overview of the rolling upgrade process. The process for upgrading Ka
...
 
 #### Step 1
 
-First, make a note of the existing protocol and log versions. Upgrade your service to the new package version `2.0.0-0.11.0.0` (with Kafka 0.11.0.0), but keep the existig protocol and log versions same by applying customized package update options. The following `options.json` file shows how protocol and log versions can be customized. Here, we assume that existing service uses 0.10.0 protocol/log versions. If you have changed protocol and log to another 0.10.x version, set them in options JSON file.
+First, make a note of the existing protocol and log versions. Upgrade your service to the new package version `1.1.27-0.11.0-beta` (with Kafka 0.11.0.0), but keep the existig protocol and log versions same by applying customized package update options. The following `options.json` file shows how protocol and log versions can be customized. Here, we assume that existing service uses 0.10.0 protocol/log versions. If you have changed protocol and log to another 0.10.x version, set them in options JSON file.
```
Comment (sascala) [inline frameworks/kafka/docs/upgrade.md]: "keep the existing protocol and log version the same by..."

File: frameworks/kafka/docs/upgrade.md [hunk 6]
```diff
@@ -65,9 +65,9 @@ First, make a note of the existing protocol and log versions. Upgrade your servi
         }
     }
 
-Now, upgrade to the package `2.0.0-0.11.0.0` that provides the new code, Kafka 0.11.0.0. 
+Now, upgrade to the package `1.1.27-0.11.0-beta` that provides the new code, Kafka 0.11.0.0.
```
Comment (sascala) [inline frameworks/kafka/docs/upgrade.md]: Now, upgrade to package `1.1.27-0.11.0-beta`, which provides the new code, Kafka 0.11.0.0.

File: frameworks/kafka/docs/upgrade.md [hunk 7]
```diff
@@ -101,25 +101,25 @@ Once you verify that all brokers are restarted, update the log version. Change t
...
      
-    $ dcos kafka update start --options=options.json 
+    $ dcos beta-kafka update start --options=options.json 
     
 The new service will be running Kafka 0.11.0.0 with two customized options (`inter.broker.protocol.version` and `log.message.format.version`). 
 
 Since protocol and log versions have been updated with new protocol and log formats, you can not directly roll back to the previous package version.
 
-**Note**: Default settings for protocol and log versions are overwritten with these customized options (steps 1 through 3), even though  their values are same as the defaults of the new package version `2.0.0-0.11.0.0` . Pay attention to these customized options for further package upgrades since they will be preserved unless overwritten explicitly.
+**Note**: Default settings for protocol and log versions are overwritten with these customized options (steps 1 through 3), even though  their values are same as the defaults of the new package version `1.1.27-0.11.0-beta` . Pay attention to these customized options for further package upgrades since they will be preserved unless overwritten explicitly.
```
Comment (sascala) [inline frameworks/kafka/docs/upgrade.md]: ^comma before "since"

**Comment by mbdcos** [comment]
This is little bit urgent.  Existing doc is misleading (explained in this PR's comments). PR#1529 has been waiting for a while and I have already seen someone complaining (old instructions will not work after 2.0.0 release). Lets remove first (less is more), then take your time to fix/update the document. Can one of you please merge PR#1553?

**Pull request merged**